registry/storage/driver/azure/azure.go

```diff
@@ -137,133 +106,140 @@ func (d *driver) PutContent(ctx context.Context, path string, contents []byte) e
 	}
-	info := blobRef.Properties
-	size := info.ContentLength
+	if props.ContentLength == nil {
+		return nil, fmt.Errorf("Failed to get ContentLength for path: %s", path)
+	}
+	size := *props.ContentLength
 	if offset >= size {
 		return io.NopCloser(bytes.NewReader(nil)), nil
 	}
 
-	resp, err := blobRef.GetRange(&azure.GetBlobRangeOptions{
-		Range: &azure.BlobRange{
-			Start: uint64(offset),
-			End:   0,
-		},
-	})
+	resp, err := blobRef.DownloadStream(ctx, &options)
 	if err != nil {
+		if is404(err) {
+			return nil, storagedriver.PathNotFoundError{Path: path}
+		}
 		return nil, err
 	}
-	return resp, nil
+	return resp.Body, nil
 }
 
 // Writer returns a FileWriter which will store the content written to it
 // at the location designated by "path" after the call to Commit.
 func (d *driver) Writer(ctx context.Context, path string, append bool) (storagedriver.FileWriter, error) {
-	blobRef := d.client.GetContainerReference(d.container).GetBlobReference(path)
-	blobExists, err := blobRef.Exists()
+	blobName := d.blobName(path)
+	blobRef := d.client.NewBlobClient(blobName)
+
+	props, err := blobRef.GetProperties(ctx, nil)
+	blobExists := true
 	if err != nil {
-		return nil, err
+		if !is404(err) {
+			return nil, err
+		}
+		blobExists = false
 	}
+
 	var size int64
 	if blobExists {
 		if append {
-			err = blobRef.GetProperties(nil)
-			if err != nil {
-				return nil, err
+			if props.ContentLength == nil {
+				return nil, fmt.Errorf("Cannot append to blob because no ContentLength property was returned for: %s", blobName)
 			}
-			blobProperties := blobRef.Properties
-			size = blobProperties.ContentLength
+			size = *props.ContentLength
 		} else {
-			err = blobRef.Delete(nil)
-			if err != nil {
+			if _, err := blobRef.Delete(ctx, nil); err != nil {
 				return nil, err
 			}
 		}
 	} else {
 		if append {
 			return nil, storagedriver.PathNotFoundError{Path: path}
 		}
-		err = blobRef.PutAppendBlob(nil)
-		if err != nil {
+		if _, err = d.client.NewAppendBlobClient(blobName).Create(ctx, nil); err != nil {
 			return nil, err
 		}
 	}
 
-	return d.newWriter(path, size), nil
+	return d.newWriter(ctx, blobName, size), nil
 }
 
 // Stat retrieves the FileInfo for the given path, including the current size
 // in bytes and the creation time.
 func (d *driver) Stat(ctx context.Context, path string) (storagedriver.FileInfo, error) {
-	blobRef := d.client.GetContainerReference(d.container).GetBlobReference(path)
+	blobName := d.blobName(path)
+	blobRef := d.client.NewBlobClient(blobName)
 	// Check if the path is a blob
-	if ok, err := blobRef.Exists(); err != nil {
+	props, err := blobRef.GetProperties(ctx, nil)
+	if err != nil && !is404(err) {
 		return nil, err
-	} else if ok {
-		err = blobRef.GetProperties(nil)
-		if err != nil {
-			return nil, err
+	}
+	if err == nil {
+		if props.ContentLength == nil || props.LastModified == nil {
+			return nil, fmt.Errorf("Invalid blob properties for blob: %s", blobName)
 		}
```

Comment:
https://github.com/golang/go/wiki/CodeReviewComments#error-strings
Which properties are invalid? Provide enough information in the error message to debug effectively.
```suggestion
		var missing []string
		if props.ContentLength == nil {
			missing = append(missing, "ContentLength")
		if props.LastModified == nil {
			missing = append(missing, "LastModified")
		}
		
		if len(missing) > 0 {	
			return nil, fmt.Errorf("required blob properties %s are missing for blob: %s", missing, blobName)
		}
```